app/src/main/java/edu/uc/reedws/musiclink/ProfileActivity.kt

```diff
@@ -0,0 +1,162 @@
+    }
+
+    /** Check for permissions to use camera*/
+    private fun prepTakePhoto() {
+        if (ContextCompat.checkSelfPermission(
+                applicationContext!!,
+                Manifest.permission.CAMERA
+            ) == PackageManager.PERMISSION_GRANTED
+        ) {
+            takePhoto()
+        } else {
+            val permissionRequest = arrayOf(Manifest.permission.CAMERA)
+            requestPermissions(permissionRequest, CAMERA_PERMISSION_REQUEST_CODE)
+        }
+    }
+
+    override fun onRequestPermissionsResult(
+        requestCode: Int,
+        permissions: Array<out String>,
+        grantResults: IntArray
+    ) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+        when (requestCode) {
+            CAMERA_PERMISSION_REQUEST_CODE -> {
+                if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    takePhoto()
+                } else {
+                    Toast.makeText(
+                        applicationContext,
+                        "Unable to take photo without permission",
+                        Toast.LENGTH_LONG
+                    ).show()
+                }
+            }
+        }
+    }
+
+    private fun takePhoto() {
+        Intent(MediaStore.ACTION_IMAGE_CAPTURE).also { takePictureIntent ->
+            takePictureIntent.resolveActivity(applicationContext!!.packageManager)?.also {
+                startActivityForResult(takePictureIntent, CAMERA_REQUEST_CODE)
+            }
+        }
+    }
+
+    /** Sets the image view to the picture the camera took*/
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (requestCode == CAMERA_REQUEST_CODE) {
+            val photoTaken = data!!.extras!!.get("data") as Bitmap
+            avatarPhotoView.setImageBitmap(photoTaken)
+        } else {
+            Toast.makeText(applicationContext, "Unable to set photo", Toast.LENGTH_LONG).show()
+        }
+    }
+
+    /** Show Dialog Screen for changing the profile name */
+    private fun showChangeNameDialog() {
+
+        val changeNameDialogBuilder = AlertDialog.Builder(this)
+        val inflater = layoutInflater
+        changeNameDialogBuilder.setTitle("Enter the new profile name")
+        val dialogLayout = inflater.inflate(R.layout.change_name_dialog, null)
+        val newName = dialogLayout.findViewById<TextView>(R.id.newProfileName)
+
+        changeNameDialogBuilder.setView(dialogLayout)
+        changeNameDialogBuilder.setPositiveButton("Done") { _, _ ->
+            txtPersonName.text = newName.text
+            val intent = Intent(this, ProfileActivity::class.java)
+            intent.putExtra("profileName", txtPersonName.text)
+        }
+        changeNameDialogBuilder.setNeutralButton("Cancel") { dialog, _ -> dialog.cancel() }
+        changeNameDialogBuilder.show()
+    }
```

Comment:
Could we make these 3 dialogs 1 function that takes a couple parameters? feels like a lot of duplicated code

Reply:
For the sake of time, I can't get this to all be one method. There's a few things that have to be referenced and I'm not sure how to set them as parameters.

Reply:
Ok sounds good